### PR TITLE
Fix issues when restoring wallets from backup

### PR DESCRIPTION
**Pull request by MrCyjaneK**
# Description

TODO

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

File: lib/core/backup_service.dart [hunk 1]
```diff
@@ -206,6 +230,16 @@ class BackupService {
...
+
+    if (!_transactionDescriptionBox.isOpen) {
+      final transactionDescriptionsBoxKey = await getEncryptionKey(secureStorage: secureStorageShared, forKey: TransactionDescription.boxKey);
+      final transactionDescriptionBox = await CakeHive.openBox<TransactionDescription>(
+        TransactionDescription.boxName,
+        encryptionKey: transactionDescriptionsBoxKey,
+      );
+      await transactionDescriptionBox.putAll(descriptionsMap);
+      return;
+    }
```
Comment (OmarHatem28) [inline lib/core/backup_service.dart:242]: why would it be closed? it gets opened in main before the app starts
and if indeed it's necessary, please use `this.secureStorage` instead

File: lib/core/backup_service.dart [hunk 2]
```diff
@@ -148,14 +157,29 @@ class BackupService {
     await _importPreferencesDump();
   }
 
+  // checked with .endsWith - so this should be the last part of the filename
+  static const ignoreFiles = [
+    "flutter_assets/kernel_blob.bin",
+    "flutter_assets/vm_snapshot_data",
+    "flutter_assets/isolate_snapshot_data",
+    ".lock",
+  ];
```
Comment (OmarHatem28) [inline lib/core/backup_service.dart:166]: out of curiosity, why do we ignore them?

File: lib/core/backup_service.dart (hunk 1 above)
Comment (OmarHatem28) [reply lib/core/backup_service.dart:242]: ok so this function `_reloadHiveWalletInfoBox` is the one that closes it, but it also closes all boxes and just open the walletInfo box, so let's make sure that's the expected behavior, cuz I don't think it is

Comment (MrCyjaneK) [reply lib/core/backup_service.dart:242]: It is an issue that [Ramo reported](https://cakewallet.slack.com/archives/C04CSQESA58/p1728607357598899?thread_ts=1728577025.568509&cid=C04CSQESA58), and I was able to reproduce later, with this fix there is no error

File: lib/core/backup_service.dart (hunk 2 above)
Comment (MrCyjaneK) [reply lib/core/backup_service.dart:166]: as for "flutter_assets/kernel_blob.bin", "flutter_assets/vm_snapshot_data", "flutter_assets/isolate_snapshot_data" if we backup/restore these files the app will crash in runtime when on debug build
and .lock files are not required - so I just included them while also adding the list - so we have even less files in backup.

**Pull request merged**